src/proto/single/mod.rs

```diff
@@ -156,6 +156,11 @@ impl Job {
     pub fn args(&self) -> &[serde_json::Value] {
         &self.args
     }
+
+    /// Data about this job's most recent failure.
+    pub fn failure(&self) -> &Option<Failure> {
+        &self.failure
+    }
```

Comment:
I think this is the minimal set of changes I wanted to include in this PR, and only did when forced to by clippy, so `Failure` is still unusable. I'm happy to commit to coming back in another PR to make `Failure` useful, or put it into this one if preferred.

Reply:
Let's do that in a separate PR then, thanks :+1: